sql/expression/in_test.go

```diff
@@ -90,7 +95,77 @@ func TestInTuple(t *testing.T) {
 			false,
 			nil,
 		},
-	}
+		{
+			name: "right values contain a different, coercible type",
+			left: expression.NewLiteral(1, sql.Uint64),
+			right: expression.NewTuple(
+				expression.NewLiteral("hi", sql.TinyText),
+				expression.NewLiteral("bye", sql.TinyText),
+			),
+			row:    nil,
+			result: false,
+		},
+		{
+			name: "right values contain a different, coercible type, and left value is zero value",
+			left: expression.NewLiteral(0, sql.Uint64),
+			right: expression.NewTuple(
+				expression.NewLiteral("hi", sql.TinyText),
+				expression.NewLiteral("bye", sql.TinyText),
+			),
+			row:    nil,
+			result: true,
```

Comment:
i ran to confirm but i still don't understand

Reply:
yup... these were fun! 😄 The string values will automatically be coerced into a number, which ends up being 0 for "hi" and "bye" since they don't contain any leading digits. That matches the left side, so this somewhat surprisingly returns true.